server/controllers/availability.js

```diff
@@ -0,0 +1,57 @@
+const asyncHandler = require("express-async-handler");
+
+const {
+  retrieveCalendar,
+  getAvailability
+} = require("../utils/googleCalendar");
+const {
+  getDayName,
+  convertToDate,
+  convertToTimeZone
+} = require("../utils/dateTime");
+
+// @route GET /availability?day=2021-07-09T05:05:09.172Z&email=acgarzeah@gmail.com
+// - day = The day we are checking if user is busy
+// - email = Email of the person we're making an appt. with
+// @desc Returns the time the user is busy and we'll
+// filter out those times on the front end
+// @access Public
+exports.getAvailability = asyncHandler(async (req, res, next) => {
+  let { day, email } = req.query;
+
+  const user = await User.findOne({ email: email });
+  const { username, availableDays, timezone, availableHours } = user;
+
+  // If user is not available on this day, then throw error
+  const isAvailable = availableDays.includes(getDayName(day, timezone));
+  if (!isAvailable) throw new Error(`${username} is not available on this day`);
+
+  // Converting the day to the user's time zone and getting the date in that time zone
+  day = convertToTimeZone(day, timezone);
+  day = new Date(day).getDate();
+
+  // Converting the user's available times to date format
+  let startISO = convertToDate(availableHours.start, day, "h:m");
+  let endISO = convertToDate(availableHours.end, day, "h:m");
+
+  // Attaching the user's time zone to their start and end time
+  startISO = convertToTimeZone(startISO, timezone);
+  endISO = convertToTimeZone(endISO, timezone);
+
+  // Accessing the user's calendar
+  const calendar = retrieveCalendar(
+    process.env.GOOGLE_CLIENT_ID,
+    process.env.GOOGLE_CLIENT_SECRET,
+    user.googleRefreshToken
```

Comment:
Shouldn't we be using their access token here as opposed to refresh? Refresh tokens are designed to retrieve new access tokens.

Reply:
Hey Ethan, at the time I was just going for a fast solution. I've been learning more about access/refresh tokens and I think authenticating using access tokens is much better for security. In the event our application gets hacked, the amount of time a hacker has to incur damage can vary drastically depending on whether an access or refresh token was leaked. I've refactored it to use access tokens now.